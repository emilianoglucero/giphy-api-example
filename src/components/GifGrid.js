import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGriditem } from './GifGriditem';

export const GifGrid = ( { category } ) => {

    const { loading, data:images } = useFetchGifs( category );


    return (
        <>
        <h3 className="animate__fadeIn animate__flash">{ category }</h3>

        { loading && <p className="animate__flash">Loading...</p> }

        <div className="card-grid"> 
            {
                images.map( img => (
                    <GifGriditem 
                        title={ img.title }
                        url={ img.url }
                        key={img.id}
                    /> 
                ))
            }
        </div>
        </>
    )
}
