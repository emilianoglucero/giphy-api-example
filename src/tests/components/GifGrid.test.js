import React from 'react';
import '../../setupTests';

import '@testing-library/jest-dom';
import { GifGrid } from '../../components/GifGrid';
import { shallow } from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe('Pruebas en el gifgrid', () => {
    

    test('debe mostrar el componenete correctamente', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const title = "One punch";
        const wrapper = shallow( <GifGrid category={ title } /> );

        expect( wrapper ).toMatchSnapshot();
    })

    test('debe mostrar items cuando se cargan imagenes useFetchGifs', () => {

        const gifs = [
            {
                id: 'ABC',
                title: 'jaja',
                url: 'jojo.com'
            }
        ] 
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const title = "One punch";
        const wrapper = shallow( <GifGrid category={ title } /> );

        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGriditem').length).toBe( gifs.length );


    })
    


})
