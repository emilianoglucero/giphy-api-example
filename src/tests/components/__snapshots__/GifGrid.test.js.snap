// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el gifgrid debe mostrar el componenete correctamente 1`] = `
<Fragment>
  <h3
    className="animate__fadeIn animate__flash"
  >
    One punch
  </h3>
  <p
    className="animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
